dashboard/app/controllers/lti_v1_controller.rb

```diff
@@ -0,0 +1,73 @@
+class LtiV1Controller < ApplicationController
+  # Don't require an authenticity token because LTI Platforms POST to this controller.
+  skip_before_action :verify_authenticity_token
+
+  # [GET/POST] /lti/v1/login(/:platform_id)
+  def login
+    if params[:client_id]
+      query_params = {client_id: params[:client_id], issuer: params[:iss]}
+    elsif params[:platform_id]
+      query_params = {platform_id: params[:platform_id]}
+    else
+      return unauthorized_status
+    end
+
+    begin
+      lti_integration = LtiIntegration.find_by!(query_params)
+    rescue ActiveRecord::RecordNotFound
+      return unauthorized_status
+    end
+
+    auth_request_body = {
+      scope: "openid",
+      response_type: "id_token",
+      client_id: lti_integration[:client_id],
+      redirect_uri: "https://studio.code.org/lti/v1/authenticate", # TODO: where can we configure this to be dynamic depending on deployment environment, i.e. prod, localhost, etc?
+      login_hint:  params[:login_hint],
+      lti_message_hint: params[:lti_message_hint] ? params[:lti_message_hint] : "", # TODO: is this needed? Canvas sends it, what happens if we don't send it back? What about other LTI Platforms?
```

Comment:
Since `nil.to_s == ''` we can simply do
```suggestion
      lti_message_hint: params[:lti_message_hint].to_s
```